grammars/typescript.cson

```diff
@@ -711,12 +711,12 @@
         'name': 'meta.brace.curly.js'
     'patterns': [
       {
-        'include': '$self'
+        'include': '$base'
       }
     ]
 
   'type':
-    'begin': '(?=)'
+    'begin': '(?=[\\(\\{\\p{L}\\p{Nl}$_])'
```

Comment:
I'm unable to understand how this code is designed to work. 

See https://regex101.com/r/qO2zF7/1

![image](https://cloud.githubusercontent.com/assets/874898/10271070/5d1615a2-6b50-11e5-92a7-de92ebcd5593.png)

What's up with matching `p` or `{L}` etc. ... or am I misreading this. Thanks for the fix btw. Perhaps all we needed was literals `(` and `{`?